core/trino-main/src/main/java/io/trino/operator/TrinoOperatorFactories.java

```diff
@@ -161,6 +166,28 @@ public OperatorFactory fullOuterJoin(
                 blockTypeOperators);
     }
 
+    @Override
+    public OutputFactory partitionedOutput(
+            TaskContext taskContext,
+            PartitionFunction partitionFunction,
+            List<Integer> partitionChannels,
+            List<Optional<NullableValue>> partitionConstants,
+            boolean replicateNullsAndAny,
+            OptionalInt nullChannel,
+            OutputBuffer outputBuffer,
+            DataSize maxPagePartitioningBufferSize)
+    {
+        return new PartitionedOutputFactory(
+                partitionFunction,
+                partitionChannels,
+                partitionConstants,
+                replicateNullsAndAny,
+                nullChannel,
+                outputBuffer,
+                maxPagePartitioningBufferSize,
+                RowBasedPagePartitioner::new);
```

Comment:
you can remove `RowBasedPagePartitioner::new`, there is a constructor that does `RowBasedPagePartitioner::new` already